### PR TITLE
STREAM-49: Migrate the build from SBT to Gradle

### PR DESCRIPTION
**Pull request by granthenke**
This patch refactors the build to use Gradle instead of SBT.
The Gradle build breaks out common functionality into separate Gradle scripts
for easier maintenance, code reuse, and sharing. There are some follow on
improvements and changes that can be made but this commit contains the
primary refactoring.

The high-level changes are:
- Moved main build functionality to the main build.gradle file
- Used the Grade Application plugin to build the zip artifact
- Moved common properties to gradle.properties
- Centralized compilation configurations in compile.gradle
- Centralized dependency versions in dependencies.gradle
- Upgraded dependency versions that didn't require non-trivial code changes
- Centralized publishing in publishing.gradle
- Centralized testing configurations in tests.gradle
- Centralized Gradle wrapper configurations/tasks in wrapper.gradle
- Added a few test configuration optimizations in tests.gradle
- Added and configured quality plugins like  scalafmt, spotless, spotbugs, and jacoco
- Centralized plugin versions in settings.gradle
- Updated copyright headers
- Updated the README
- Fixed testScriptCommand_mandatory_config
- Fixed "ALLOWED SCHEMA CHANGES" sort issue
- Removed unused lib jars
- Removed unused scripts and make files

The majority of the Java code changes are spotless auto-formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,11 +7,12 @@ See the [phData Streamliner User Documentation site](https://docs.customer.phdat
...
 
 ## Packaging Streamliner
-Streamliner uses [SBT](https://www.scala-sbt.org/) as a dependency management and build tool.  [SBT Native Packager](https://www.scala-sbt.org/sbt-native-packager/) attempts to make building packages for different operating systems easier.  Streamliner includes SBT Native Packager as a SBT plugin and should be used when packaging Streamliner.
+Streamliner uses [Gradle](https://gradle.org/) as a dependency management and build tool.
+The [Gradle Application Plugin](https://docs.gradle.org/current/userguide/application_plugin.html) attempts to make building packages for different operating systems easier.
 
 ### Creating Streamliner Package
-1. Execute `sbt clean universal:packageBin`
-2. Copy zip from `target/universal/streamliner-<version>.zip` to intended install directory
+1. Execute `./graldew clean assemble`
```
Comment (akuma432) [inline README.md:14]: typo

**Comment by akuma432** [comment]
Since we deleted https://github.com/phdata/streamliner/tree/master/lib, we should add dependency for them. I was able to connect to oracle after adding `'com.oracle.database.jdbc:ojdbc10:19.12.0.0'` in dependecies.gradle. Similarly we should add for other libs.

**Comment by akuma432** [comment]
Initially when I pulled these changes I was getting error `package edu.umd.cs.findbugs.annotations does not exist` after running any gradlew command. 
I was able to fix that issue after setting `lombok.extern.findbugs.addSuppressFBWarnings` to false in lombok.config or adding dependency `'com.google.code.findbugs:annotations:3.0.1'`.  
@granthenke  did you also face this issue or I missed something?

**Comment by akuma432** [comment]
I created streamliner package using the gradle command mentioned in README.md and tested streamliner functionalities. I had to add `ojdbc` dependency as I mentioned in my previous comment as well. After that I did not face any issue. Also all Junit cases are passing.

**Comment by granthenke** [comment]
Thanks for the review @akuma432. I pushed an update that should address the issues found.

**Comment by akuma432** [comment]
LGTM

**Review by akuma432 — APPROVED** [review]

**Comment by brockn** [comment]
Nice work. LGMT

**Review by brockn — APPROVED** [review]

**Pull request merged**